### ISSUE TITLE
Environment specification too strict

### ISSUE THREAD

**Issue by jmrgonza** [EC-Earth/ece2cmor3#193]
Hi, I am trying to create the ece2cmor3 conda environment following the instructions in README.md, but it fails to find some dependencies. I think that the reason is that some dependencies that are specified with an exact build string (after the second = sign) in the environment.yml file, are not currently published in the conda channels.
Also, I think that the environment.yml lists a lot of packages that are probably not required for this project, and it would be good to do some cleaning to simplify the installation.

**Comment by treerink** [comment]
Hi, you are right. However, I just checked and I am able to create the ece2cmor3 environment if I remove in the `environment.yml` the line 
`````scripts
libiconv=1.15=0
`````
I will commit this fix today.

Actually we tried yesterday to reduce the number of non-necessary dependencies, but unfortunately this is slightly more subtle. At this moment we hope to be able to change to cmor-3.3.3, but see https://github.com/PCMDI/cmor/issues/366 and here #190 and #187. 

In general the trouble is to find the correct dependencies for hdf5 1.10 and netcdf >4.4 in order to prevent conflicts between the dependencies of the packages in ece2cmor3.

**Comment by treerink** [comment]
Creating the `ece2cmor3` environment should work again out of the box (tested on linux).

**Comment by jmrgonza** [comment]
Hi Thomas,
is it possible that you have some cache copies of packages installed in the past? The first dependency that triggered an error in my environment creation was hdf5=1.10.1=h9caa474_1. If I do a 
> conda search hdf5
I do not get any result with the h9caa474_1 build string.
After I erased the build string from the package specification, other packages showed similar problems for me. After cleaning all of them in a similar fashion, it complains about the intel-openmp 2018.0.0 package, which I doubt that it is necessary for ece2cmor3.

**Comment by treerink** [comment]
Hmm strange because I get it:
`````scripts
conda search hdf5

Fetching package metadata .............
hdf5                         1.8.9                         0  defaults        
                             1.8.9                         1  defaults        
                             1.8.13                        0  defaults        
                             1.8.13                        1  defaults        
                             1.8.14                        0  defaults        
                             1.8.15.1                      0  defaults        
                             1.8.15.1                      1  defaults        
                             1.8.15.1                      2  defaults        
                             1.8.15.1                      3  defaults        
                             1.8.16                        0  defaults        
                             1.8.17                        0  defaults        
                             1.8.17                        1  defaults        
                             1.8.17                        2  defaults        
                             1.8.18               h525d4c3_0  defaults        
                             1.8.18               h6792536_1  defaults        
                             1.10.1               hb0523eb_0  defaults        
                          *  1.10.1               h9caa474_1  defaults        
                             1.10.2               hba1933b_1  defaults
`````
And, by the way, last week I also was able to create the ece2cmor3 environment on a ubuntu laptop where I had not installed it at all ever before.

**Comment by jmrgonza** [comment]
It looks like the list of available packages depends on the system making the request. This is my output:
```shell
(root) jmrodriguez@chinyero:~/ecearth/ece2cmor3$ conda search hdf5
Fetching package metadata .............
hdf5                         1.8.9                         0  defaults
                             1.8.9                         1  defaults
                             1.8.13                        0  defaults
                             1.8.13                        1  defaults
                             1.8.14                        0  defaults
                             1.8.15.1                      0  defaults
                             1.8.15.1                      1  defaults
                             1.8.15.1                      2  defaults
                             1.8.15.1                      3  defaults
                             1.8.16                        0  defaults
                             1.8.17                        0  defaults
                             1.8.17                        1  defaults
                             1.8.17                        2  defaults
                             1.8.16                        3  conda-forge
                             1.8.16                        4  conda-forge
                             1.8.17                        0  conda-forge
                             1.8.17                        2  conda-forge
                             1.8.17                        3  conda-forge
                             1.8.17                        4  conda-forge
                             1.8.17                        5  conda-forge
                             1.8.17                        6  conda-forge
                             1.8.17                        7  conda-forge
                             1.8.17                        8  conda-forge
                             1.8.17                        9  conda-forge
                             1.8.17                       10  conda-forge
                             1.8.17                       11  conda-forge
                             1.8.18                        0  conda-forge
                             1.8.18                        1  conda-forge
                             1.8.18                        2  conda-forge
                             1.8.18                        3  conda-forge
                             1.8.19                        0  conda-forge
                             1.8.19                        1  conda-forge
                             1.8.19                        2  conda-forge
                             1.8.20                        0  conda-forge
                             1.8.20                        1  conda-forge
                             1.10.1                        0  conda-forge
                             1.10.1                        1  conda-forge
                             1.10.1                        2  conda-forge
                             1.10.2                        0  conda-forge
```
I will check my configuration, but I wonder if people using different platforms or anaconda versions will be able to meet these dependencies. Do you think that ```ece2cmor3``` won't work if only the package versions, and not the build strings, are listed in the requirements?

**Comment by treerink** [comment]
Hi Jose,

Could you try the env-cleanup branch? There we put a strongly simplified `environment.yml`, at our linux it is working, but we want to know whether this cmor-3.3.3 "nigtly" built works everywhere.

**Comment by jmrgonza** [comment]
Hi  Thomas,
this is the output with the `env-cleanup` branch:
```shell
(root) jmrodriguez@chinyero:~/ecearth/ece2cmor3$ conda env create -f environment.yml
Fetching package metadata .............

NoPackagesFoundError: Package missing in current linux-64 channels:
  - cmor 2018.06.28.master.npy1.14 py27h9ac9557_0

(root) jmrodriguez@chinyero:~/ecearth/ece2cmor3$ conda search cmor
Fetching package metadata .............
cmor                         3.3.2            py27h470a237_0  conda-forge
                             2.9.2                     PCMDI  pcmdi
                             3.0beta1                  PCMDI  pcmdi
                             3.0.2                     PCMDI  pcmdi
                             3.0.3                     PCMDI  pcmdi
                             3.0.4                     PCMDI  pcmdi
                             3.0.5                     PCMDI  pcmdi
                             3.0.6                     PCMDI  pcmdi
                             3.1.0                     PCMDI  pcmdi
                             3.1.1                     PCMDI  pcmdi
                             3.1.2                     PCMDI  pcmdi
                             3.1.3                     PCMDI  pcmdi
                             3.2.0               np110py27_0  pcmdi
                             3.2.0               np111py27_0  pcmdi
                             3.2.0                np19py27_0  pcmdi
                             3.2.1               np110py27_0  pcmdi
                             3.2.1               np111py27_0  pcmdi
                             3.2.1                np19py27_0  pcmdi
                             3.2.2               np110py27_0  pcmdi
                             3.2.2               np111py27_0  pcmdi
                             3.2.2               np112py27_0  pcmdi
                             3.2.2                np19py27_0  pcmdi
                             3.2.3               np110py27_0  pcmdi
                             3.2.3               np111py27_0  pcmdi
                             3.2.3               np112py27_0  pcmdi
                             3.2.3                np19py27_0  pcmdi
                             3.2.4               np110py27_0  pcmdi
                             3.2.4               np111py27_0  pcmdi
                             3.2.4               np112py27_0  pcmdi
                             3.2.4                np19py27_0  pcmdi
                             3.2.5               np110py27_0  pcmdi
                             3.2.5               np111py27_0  pcmdi
                             3.2.5               np112py27_0  pcmdi
                             3.2.5                np19py27_0  pcmdi
                             3.2.6               np110py27_0  pcmdi
                             3.2.6               np111py27_0  pcmdi
                             3.2.6               np112py27_0  pcmdi
                             3.2.6               np113py27_0  pcmdi
                             3.2.6                np19py27_0  pcmdi
                             3.2.7               np110py27_0  pcmdi
                             3.2.7               np111py27_0  pcmdi
                             3.2.7               np112py27_0  pcmdi
                             3.2.7               np113py27_0  pcmdi
                             3.2.7                np19py27_0  pcmdi
                             3.2.8            py27h321e43f_0  pcmdi
                             3.2.8            py27h5f41edd_0  pcmdi
                             3.2.8            py27had5b656_0  pcmdi
                             3.2.8            py27hca59ec5_0  pcmdi
                             3.2.8            py27hfbdf3cc_0  pcmdi
                             3.3.0            py27hd7ebaad_0  pcmdi
                             3.3.0            py27he7c6f3b_0  pcmdi
                             3.3.0            py27hefa619b_0  pcmdi
                             3.3.1.npy1.11    py27h49459c2_0  pcmdi
                             3.3.1.npy1.12    py27h82befd7_0  pcmdi
                             3.3.1.npy1.13    py27h16be09d_0  pcmdi
                             3.3.2.npy1.12    py27h85eb232_0  pcmdi
                             3.3.2.npy1.13    py27hb8bc26d_0  pcmdi
                             3.3.3.npy1.11    py27h481b005_0  pcmdi
                             3.3.3.npy1.12    py27h69563bd_0  pcmdi
                             3.3.3.npy1.13    py27h81ec5f0_0  pcmdi
                             3.3.3.npy1.14    py27ha5dfbcb_0  pcmdi
```
I will start again with a new miniconda installation, to make sure that I do not have outdated repository urls in some config file.

**Comment by jmrgonza** [comment]
Still with my existing installation, I was able to get all the conda dependencies by changing the list of channels in the `env-cleanup` branch to:
```
channels:
- pcmdi/label/nightly
- conda-forge
- defaults
- anaconda
```
I am still getting some error with the `f90nml`  dependency listed in the `pip`section, but the environment was created and the error that I get points some network problem with pip (maybe due to my proxy setting).

**Comment by treerink** [comment]
Yes you are right the
`````scripts
- pcmdi/label/nightly
`````
has to be added. I had trouble to repeat it at another linux machine as well. The
`````scripts
- anaconda
`````
I am not sure, will test  it.

**Comment by treerink** [comment]
Solved, see closing comment in #190
